scripts/linux/electron-installer-appimage.sh

```diff
@@ -102,17 +101,14 @@ else
   exit 1
 fi
 
-if [ -z "$TMPDIR" ]; then
-  TMPDIR=$(mktemp -d)
-fi
-mkdir -p "$TMPDIR"
+APPIMAGE_TMPDIR=$(mktemp --dry-run -d -t "$ARGV_APPLICATION_NAME-linux-$ARGV_ARCHITECTURE-XXXX")
```

Comment:
> I'll create a generic script that performs downloads (and takes a MD5 argument to verify the files), tries to cache if necessary, and will make all scripts call it.

Awesome. :tada: 

And yeah, the BuildRoot Makefiles are ridiculously huge and complicated - I think they even go as far as using data-files to then auto-generate (and execute) other makefiles - proper mind-bending stuff ;-)

Do you want me to approve this PR, or will it be superseded by your centralised-downloads PR?